include/circt/Dialect/LLHD/IR/LLHD.td

```diff
@@ -35,6 +35,27 @@ def LLHD_Dialect : Dialect {
   let hasConstantMaterializer = 1;
 }
 
+// Begin Copy-Pasted HWDialect declaration
+def HWDialect : Dialect {
```

Comment:
Could `include "circt/Dialect/HW/HW.td"` work here?

Reply:
In that case all the HW ops are also generated in `LLHD.h.inc`. When both `LLHDOps.h` and `HWOps.h` are then included in some file a lot of redefinition errors are thrown and a bunch or other errors.

Reply:
Maybe something like a more specific `include "[...]HWTypes.td"` could work? Not sure if that is setup yet. There's some duplication of HW in the SV/FIRRTL dialects as well, which is sort of ugly. Pretty sure some refactoring of type constraints and the like into "includable" files could work.

Reply:
Hmm, I'm not exactly sure what you mean. Could you maybe expand a bit more on how that refactoring would look like?

Reply:
Currently all the C++ code for the HW types is also generated in the LLHD directory but I just ignore it and include `HWTypes.h`. So essentially what I need is only the definitions in `HWTypes.td` that inherit from `DialectType<HWDialect, [...]`.  Moving them into a separate file and including it in `HWTypes.td` and `LLHD.td` would work if we replace `DialectType<HWDialect, [...]` with `Type<[...]`, but then they aren't linked to the HW dialect anymore, right? Unfortunately, I'm not familiar enough to immediately tell what the consequences of this change are. Is that maybe the refactoring you were thinking of?

Reply:
Sorry, my comment above really wasn't helpful 😛.

After @mikeurbach's comment you noted that `include "circt/Dialect/HW/HW.td"` would make all the HW ops appear as part of the generated LLHD files. So I thought maybe there's a different file we could include, like `include "circt/Dialect/HW/HWTypes.td"`, which only contains types.

That probably won't work because the types need the `HWDialect` to be defined. But that lives in `HW.td` and pulls in all the other files. One thing that might work is to move the `HWDialect` definition from `HW.td` into a separate `HWDialect.td`, such that you can include `HWTypes.td` (which could include `HWDialect.td`) form LLHD without pulling in all the ops and other stuff.

On the other hand: if you have to start to do serious refactoring/restructuring of code in the HW dialect, it might be worthwhile to postpone that for a separate PR that is dedicated to some restructuring. Then people can chime in more easily because the only change is the refactoring.

Reply:
On the other hand, if all that's needed is moving `HWDialect` into its own `HW/HWDialect.td` file and including that here and in `HW/HW.td`, then that might be worth doing 😃